CharlieBackend.Business/Services/ScheduleServiceFolder/ScheduleService.cs

```diff
@@ -112,6 +112,30 @@ public async Task<Result<EventOccurrenceDTO>> DeleteScheduleByIdAsync(long id, D
             return Result<EventOccurrenceDTO>.GetSuccess(_mapper.Map<EventOccurrenceDTO>(eventOccurrenceResult));
         }
 
+        public async Task<Result<bool>> DeleteConcreteScheduleByIdAsync(long eventId)
+        {
+            string error = await ValidateScheduledEventId(eventId);
+
+            if (error != null)
+            {
+                return Result<bool>.GetError(ErrorCode.ValidationError, error);
+            }
+
+            var eventEntity = await _unitOfWork.ScheduledEventRepository.GetByIdAsync(eventId);
+
+            if (eventEntity != null)
+            {
+                var mappedEvent = _mapper.Map<ScheduledEventDTO>(eventEntity);
+                await _unitOfWork.ScheduledEventRepository.DeleteAsync(eventId);
+                await _unitOfWork.CommitAsync();
+
+                return Result<bool>.GetSuccess(true);
```

Comment:
if concrete chedule is already disabled false should be returned. Please, add propper check

Reply:
No, when we delete the schedule which was deleted before we never get it again. I was so amazed by filters because of it. We never get result with the schedule, which has property is_deleted as true
![Screenshot_25](https://user-images.githubusercontent.com/56502713/110365430-b5734680-804d-11eb-9149-c6e33b4c799c.png)

Reply:
I added a filter here to ApplicationContext. It checks entities and returns us one SchedulEdevents with the field IsDeleted as false. It allows us to avoid many checks of ScheduledEvents while making requests to EventOccurrences, concrete ScheduledEvents, and other entities.
![Screenshot_3](https://user-images.githubusercontent.com/56502713/110465850-a045f880-80dd-11eb-82c5-5f5cbbf7552b.png)